compiler/damlc/tests/src/DA/Test/Repl.hs

```diff
@@ -62,6 +62,44 @@ main = do
                   , input "debug x"
                   , matchOutput "^.*: 2$"
                   ]
+            , testInteraction' "parse error"
+                  [ input "eaiu\\1"
+                  , matchOutput "^parse error.*$"
+                  , input "debug 1"
+                  , matchOutput "^.*: 1"
+                  ]
+            , testInteraction' "unsupported statement"
+                  [ input "(x, y) <- pure (1, 2)"
+                  , matchOutput "^Unsupported statement:.*$"
+                  , input "debug 1"
+                  , matchOutput "^.*: 1"
+                  ]
+            , testInteraction' "type error"
+                  [ input "1"
```

Comment:
I think it would be more intuitive if a raw, pure expression did not result in a type error, but instead caused a `debug` to be inserted automatically, turning it into a `Script _` expression. I think we should try to reserve "type errors" in the repl for non-pure expressions (e.g. tried to use Update monad instead of Script monad), or for pure expressions that have type errors in them (e.g. passed an argument of the wrong type). This is way more work than should be in this PR, though.

Reply:
(As for how to implement that, we could catch type errors with "expected Script", add a `debug`, and try again. Maybe that's kinda messy, but I think it would make the interface more accessible.)

Reply:
I fully agree that this would be a nicer UX! GHCi also prints the result if you have something of type `IO a` where `a` is an instance of `Show` (and not `()) which would be nice to have. It also has some magic defaulting to `IO` which makes things like `pure 1` work automagically (that’s the main reason why I currently have the type annotation to `Script _`. Not sure how much more clever that is than typechecking twice and if it’s possible to adapt this for our usecase.